src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-filter/GeneFilter.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { RootState } from 'src/store';
+import { isEntityViewerSidebarOpen } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarSelectors';
+
+import { ReactComponent as CloseIcon } from 'static/img/shared/close.svg';
+import { ReactComponent as ChevronUp } from 'static/img/shared/chevron-up.svg';
+import RadioGroup, {
+  RadioOptions
+} from 'src/shared/components/radio-group/RadioGroup';
+
+import styles from './GeneFilter.scss';
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+
+type Props = {
+  toggleFilter : () => void ;
+  isSidebarOpen : boolean;
+};
+
+type OptionValue = string | number | boolean;
+
+const radioData: RadioOptions = [
+  { value: 'default', label: 'Default' },
+  { value: 'length_longest', label: 'Spliced length: longest - shortest' },
+  { value: 'length_shortest', label: 'Spliced length: shortest - longest' }
+];
+
+const GeneFilter = (props: Props) => {
```

Comment:
The sorting and filtering options are applied to the transcripts of a gene, not to the gene itself. Doesn't this make the name GeneFilter misleading? Wouldn't TranscriptsSortAndFilter be more appropriate?

Reply:
The way i was looking at it is the component is being used in the Gene View hence Gene filter.  I guess it does make sense the other way round where the filters are applied to the transcript, i wasn't looking it at this way yet but makes sense more. But i think i will keep it short to TranscriptsFilter